app/views/AdminView/components/VariantText/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, {
+  useState, useEffect, useCallback,
+} from 'react';
+import { CircularProgress } from '@mui/material';
+import { VariantTextType } from '@/common';
+import IPRWYSIWYGEditor from '@/components/IPRWYSIWYGEditor';
+import sanitizeHtml from 'sanitize-html';
+import api from '@/services/api';
+import snackbar from '@/services/SnackbarUtils';
+import DataTable from '@/components/DataTable';
+import columnDefs from './columnDefs';
+import AddEditVariantText from './components/AddEditVariantText';
+
+function VariantText(): JSX.Element {
+  const [variantText, setVariantText] = useState<VariantTextType[]>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isAdding, setIsAdding] = useState(false);
+  const [editingData, setEditingData] = useState(null);
+
+  // Grab template appendices
+  useEffect(() => {
+    let cancelled = false;
+    const getVariantText = async () => {
+      try {
+        const variantTextResp = await api.get('/variant-text').request();
+        if (variantTextResp) {
+          const sanitizedVariantText = variantTextResp.map((variant: VariantTextType) => ({
+            ...variant,
+            text: sanitizeHtml(variant.text, {
+              allowedSchemes: [],
+              allowedAttributes: {
+                '*': ['style'],
+              },
+            }),
+          }));
+          if (!cancelled) {
+            setVariantText(sanitizedVariantText);
+          }
+        }
+      } catch (err) {
+        snackbar.error(`Network error: ${err}`);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    getVariantText();
+    return function cleanup() { cancelled = true; };
+  }, []);
+
+  const handleOnEdit = useCallback((rowData) => {
+    setEditingData(rowData);
+    setIsEditing(true);
+  }, []);
+
+  const handleOnAdd = useCallback(() => {
+    setIsAdding(true);
+  }, []);
+
+  const handleOnDelete = useCallback(async (rowData: VariantTextType) => {
+    try {
```

Comment:
Visually, looks good, there is one interaction missing though

When a user deletes a record on VariantText main view, the table doesn't get updated 

use the [Appendix onDelete as reference](https://github.com/bcgsc/pori_ipr_client/blob/0e8d55d7c37515a3f75cac26c6ca057b8cd33557/app/views/AdminView/components/Appendices/index.tsx#L60) to update the `variantText` state

Reply:
just pushed

Reply:
rats, too slow